### PR TITLE
Improve sonos.py a bit

### PR DESCRIPTION
**Pull request by lingfish**
Some modifications to better handle grouped speakers, as well as showing them as a target choice.

### CODE REVIEW

File: share/gpodder/extensions/sonos.py [hunk 1]
```diff
@@ -25,38 +25,35 @@
 
 SONOS_CAN_PLAY = lambda e: 'audio' in e.file_type()
 
-
```
Comment (elelay) [inline share/gpodder/extensions/sonos.py]: keep 2 blank lines between top level items (pycodestyle)

File: share/gpodder/extensions/sonos.py [hunk 2]
```diff
@@ -25,38 +25,35 @@
...
-                speaker_ip = controller.ip_address
+                self.speakers[speaker.uid] = speaker
 
-    def _stream_to_speaker(self, speaker_ip, episodes):
+    def _stream_to_speaker(self, speaker_uid, episodes):
         """ Play or enqueue selected episodes """
 
         urls = [episode.url for episode in episodes if SONOS_CAN_PLAY(episode)]
-        logger.info('Streaming to Sonos %s: %s' % (speaker_ip, ', '.join(urls)))
+        logger.info('Streaming to Sonos %s: %s'%(self.speakers[speaker_uid].ip_address, ', '.join(urls)))
```
Comment (elelay) [inline share/gpodder/extensions/sonos.py]: add space before and after modulo operator (pycodestyle)

File: share/gpodder/extensions/sonos.py [hunk 3]
```diff
@@ -25,38 +25,35 @@
...
                 continue
 
             name = info.get('zone_name', None)
+            uid = speaker.uid
 
             # devices that do not have a name are probably bridges
             if name:
-                self.speakers[speaker_ip] = name
-                speaker_ip = controller.ip_address
+                self.speakers[speaker.uid] = speaker
```
[the remark this answers is not in the document]
Comment (lingfish) [reply share/gpodder/extensions/sonos.py]: Nice catch.

File: share/gpodder/extensions/sonos.py (hunk 1 above)
Comment (lingfish) [reply share/gpodder/extensions/sonos.py]: This must have been a mis-paste.

File: share/gpodder/extensions/sonos.py (hunk 2 above)
Comment (lingfish) [reply share/gpodder/extensions/sonos.py]: Well, technically, modulo's shouldn't be used anymore from Python 3; `.format` [should be](https://docs.python.org/2/library/stdtypes.html#str.format).  Fixed though.

**Comment by lingfish** [comment]
All done.

**Comment by elelay** [comment]
Merged, thanks!

**Pull request merged**